docs: add more meat to the development section (#1801)
urbica/martin: docs/README.md

# Martins documentation

> Martins documentation is available at <https://maplibre.org/martin>.

Reviewer: ```suggestion
Martin documentation is available at <https://maplibre.org/martin>.
```

Reviewer 2: Or Martin's.? (I might be wrong, English is not my native language)

Author: Let's actually be more explicit in this

```suggestion
The documentation of `maplibre/martin` is available at <https://maplibre.org/martin>.
```

To build/develop this documentation locally, you can install `mdbook` and `mdbook-linkcheck`.

> This can be done by running the following commands:

Reviewer 2: ```suggestion
This can be done by running the following command:
```
Or just remove it?

```suggestion

```

```bash
cargo install mdbook
```

## Development

You can simply edit the markdown files in the `src` directory and run the following command (from the project root directory) to preview the changes:

```bash
mdbook watch --open docs
```

Next to showing you how the docs will look, this also runs a link checker to ensure that all links are valid.

> [!TIP]
> Make sure that all pages are linked from [`src/SUMMARY.md`](src/SUMMARY.md).
> **Only** pages linked will be rendered.

> > See the mdbook documentation for more information.

Reviewer 2: ```suggestion   
> See the [mdbook documentation](https://rust-lang.github.io/mdBook/) for more information.
```
